main.tf

```diff
@@ -1,13 +1,69 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled      = module.this.enabled
+  cluster_name = join("", aws_ecs_cluster.default.*.name)
+
+  capacity_providers_fargate = [for name, is_enabled in {
+    FARGATE : var.capacity_providers_fargate,
+    FARGATE_SPOT : var.capacity_providers_fargate_spot
+  } : name if is_enabled]
+
+  capacity_providers = distinct(concat(
+    [for key, value in aws_ecs_capacity_provider.ec2 : value.name],
+    local.capacity_providers_fargate
+  ))
+
+  default_capacity_strategy = [
+    for name, weight in var.default_capacity_strategy.weights : {
+      capacity_provider = name,
+      weight            = weight,
+      base              = var.default_capacity_strategy.base.provider == name ? var.default_capacity_strategy.base.value : null
+    }
+  ]
+}
+
+resource "aws_ecs_cluster" "default" {
+  count = local.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  name = module.this.id
+
+  setting {
+    name  = "containerInsights"
+    value = var.container_insights_enabled ? "enabled" : "disabled"
   }
+
+  configuration {
+    execute_command_configuration {
+      kms_key_id = var.kms_key_id
+      logging    = var.logging
+      dynamic "log_configuration" {
+        for_each = var.logging == "OVERRIDE" ? [var.log_configuration] : []
+        content {
+          cloud_watch_encryption_enabled = log_configuration.value["cloud_watch_encryption_enabled"]
+          cloud_watch_log_group_name     = log_configuration.value["cloud_watch_log_group_name"]
+          s3_bucket_name                 = log_configuration.value["s3_bucket_name"]
+          s3_bucket_encryption_enabled   = log_configuration.value["s3_bucket_encryption_enabled"]
+          s3_key_prefix                  = log_configuration.value["s3_key_prefix"]
+        }
+      }
+    }
+  }
+
+  tags = module.this.tags
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_ecs_cluster_capacity_providers" "default" {
+  count = local.enabled ? 1 : 0
```

Comment:
What if a capacity provider isn't desired ? I think an additional input would be good to have here